src/suites/cts/validation/queue_submit.spec.ts

```diff
@@ -0,0 +1,42 @@
+export const description = `
+queue submit validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+export const g = new TestGroup(ValidationTest);
+
+g.test('submitting with a mapped buffer is disallowed', async t => {
+  // Create a map-write buffer
+  const buffer = t.device.createBuffer({
+    size: 4,
+    usage: GPUBufferUsage.MAP_WRITE | GPUBufferUsage.COPY_SRC,
+  });
+
+  // Create a fake copy destination buffer
+  const targetBuffer = t.device.createBuffer({
+    size: 4,
+    usage: GPUBufferUsage.COPY_DST,
+  });
+
+  const commandEncoder = t.device.createCommandEncoder();
+  commandEncoder.copyBufferToBuffer(buffer, 0, targetBuffer, 0, 4);
+
+  // Submitting when the buffer has never been mapped should succeed
+  const commandBuffer = commandEncoder.finish();
```

Comment:
This command buffer gets reused 3 times, which is not supposed to be valid (though I think it's currently valid in chrome). Need e.g. a helper to re-create the command buffer.

Reply:
Done.